.github/workflows/full-mpich.yml

```
@@ -0,0 +1,118 @@
+# Full version of FreeFem, using MPICH installed by PETSc
+name: Full version MPICH
+
+on:
+  push:
+    branches:
+      - develop
+      - master
+      
+  pull_request:
+    branches:
+      - develop
+
+# For development phase
+# on:
+#   workflow_dispatch:
+  
+jobs:
+  linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [22.04, 24.04]
+        cfg:
+        - {opts: --enable-debug}
+        - {opts: --enable-optim --enable-generic}
+
+    name: Ubuntu ${{ matrix.version }} with ${{ matrix.cfg.opts }} \
+            - ${{ github.workflow }}
+            
+    runs-on: ubuntu-${{ matrix.version }}
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Install
+        run: |
+          sudo apt update
+          sudo apt install gcc g++ gfortran m4 patch git libblas-dev wget cmake \
+          liblapack-dev libhdf5-dev libgsl-dev flex bison autoconf automake \
```

```suggestion
          sudo apt install gcc g++ gfortran m4 patch git wget cmake \
          libhdf5-dev libgsl-dev flex bison autoconf automake \
```
PETSc will install OpenBLAS automatically.